### ISSUE TITLE
Integration of libreSpeed + speed test files, public

### ISSUE THREAD

**Issue by MagnaCapax** [MagnaCapax/PMSS#4]
These should be added via the main nginx installation, in a manner anyone can use them.

Some restrictions and limits needs to be put in place however to avoid abuse / DOS type behavior.

**Comment by MagnaCapax** [comment]
For public use path is /var/www

**Comment by MagnaCapax** [comment]
## Investigation Report

**Issue:** #4 — Integration of libreSpeed + speed test files, public
**Investigated:** 2026-05-30
**Status:** PARTIAL

### Summary
A public download speed-test file already ships today: `update-step2.php` generates `/var/www/testfile` (100 MB) on every update, served by the main nginx default site. The interactive **LibreSpeed** app is **not** present, and the **abuse/DOS limits this issue explicitly asks for are not applied** to the public path (the `limit_conn_zone` is declared but never used on `/var/www`). The cleanest path is to **split** this issue: land the small, ADR-aligned rate-limit fix now, and treat the LibreSpeed web app as a separate, gated decision.

### What The Issue Asks For
1. Integrate **LibreSpeed** (interactive browser speed test) via the main nginx installation.
2. Provide **downloadable speed test files**, public, under `/var/www`.
3. Add **restrictions/limits** to prevent abuse / DOS behavior.

### Implementation Status
- **Download test file — PARTIALLY present.** `scripts/util/update-step2.php` (the "Ensure the standard download speed test file exists" step) creates `/var/www/testfile` as a 100 MB random file and regenerates it if missing or the wrong size. The main nginx default site (`etc/seedbox/config/template.nginx-site-default`) has `root /var/www` and listens on `:80` and `:443` as `default_server`, so the file is publicly reachable. This satisfies the spirit of "speed test files, public" at a basic level and confirms `/var/www` is the correct location (per the issue comment).
- **LibreSpeed — not present.** No LibreSpeed assets or backend exist in the tree. The main nginx has **no PHP/FastCGI** wiring (PHP is handled only by per-user lighttpd); LibreSpeed's optional PHP backend would therefore require adding a PHP execution path to the public front door.
- **Abuse/DOS limits — not applied to the public path.** `template.nginx-site-default` declares `limit_conn_zone $binary_remote_addr zone=addr:10m;`, but the default `location /` (serving `/var/www`, including `testfile`) applies **no** `limit_conn`/`limit_rate`. Those directives exist only in `etc/seedbox/config/template.nginx-user` (per-user proxy vhosts). There is no global `limit_req` in `template.nginx-conf`. Net effect: the existing public 100 MB file is served today with no connection or rate cap.

### What's Missing
1. Connection/rate limits on the public download asset (the requested DOS protection).
2. The LibreSpeed interactive UI (upload test, ping/jitter, IP/ISP info) — none of which the static file provides.
3. Any documentation/discoverability of the existing `testfile` URL.

### Mission Alignment
- **Reliability:** keeping the main nginx a thin front door (no new PHP backend) and using nginx-native limits is the conservative, low-failure-mode option.
- **Value for money / transparency:** a provider-hosted "test your speed to our datacenter" endpoint is a legitimate pre-sales/diagnostic asset; the static file already delivers the core download signal at zero new surface.
- **Requirements check:** the public Network Diagnostics wiki frames browser speed tests as *"inherently unreliable"* and steers users to iperf3 / MTR / network-speed.xyz. Worth weighing before investing in a branded LibreSpeed UI — that is the published stance.

### Devil's Advocate
**Why not build LibreSpeed as written?** The main nginx is intentionally minimal; a PHP-backed speed test adds an executable surface on the public front door plus a per-server maintenance burden. A public, unauthenticated download/upload generator is itself an abuse vector that must be rate-limited regardless, and the static file already answers "how fast can I pull from this server?" for most visitors. **Partial rebuttal:** LibreSpeed adds upload + latency/jitter and a friendly UI a raw file cannot; if a marketing-grade public surface is wanted, that value is real — but it should be delivered backendless/static and behind review, not bolted onto nginx.

### Quality / risk checks
- The rate-limit slice addresses a **present, live** exposure (an already-public unlimited 100 MB download), not a speculative optimization.
- Status is PARTIAL with file-cited evidence; no completion is claimed.
- The primary recommendation is actionable (a concrete config slice), not a deferral; only the marketing-vs-cost product call is left to maintainers.
- Recommended limits are **stateless** nginx caps (no lockout failure mode), consistent with the project's decision not to enable stateful banning by default.
- Every cited file/ADR was read during this investigation.

### Related Incidents & Data
No recurring operational pain is associated with this 2-year-old enhancement. The public download file has existed since the project's earliest commits without a reported abuse incident — so the DOS-limit gap is a latent exposure to harden, not an active fire. Lower urgency, but the requested limit is still the correct, cheap hardening.

### Cross-Issue Analysis
- **#51 (Variety of automatic Benchmarks after installation):** adjacent but **distinct**. #51 is *internal, outbound* QC (yabs.sh / network-speed.xyz / fio, logged for provisioning decisions). #4 is a *public, inbound* test (a visitor measuring their link to the server). Different audience and direction → **not a merge**; they could one day share a results-display surface.
- No duplicate of #4 found. #45 / #501 concern internal per-user I/O metrics, unrelated to a public speed test.

### Dependencies & Blockers
- The rate-limit slice has **no blockers** — the `addr` zone is already declared; it just needs to be applied.
- The LibreSpeed slice is blocked on a product decision (is the public marketing surface wanted?) and, if yes, on the same gate the project applied to the browser-console request (#326): packaging in the supported Debian baselines, security review, and a per-user-lighttpd proxy contract — or static/backendless delivery with no backend at all.

### Reuse, Refactor & Test Strategy
- **Existing reuse points:** `limit_conn_zone addr` (already declared in `template.nginx-site-default`); the `limit_rate_after`/`limit_rate`/`limit_conn addr` pattern already proven in `template.nginx-user`; `/var/www` static serving already in place; `update-step2.php` already owns the testfile lifecycle.
- **Refactor opportunity:** add a dedicated `location = /testfile` block in the default-site template(s) carrying the limits, so the public-asset policy is explicit and testable rather than inheriting the unlimited `location /`. Keep `template.nginx-site-default` and `template.nginx-site-default-ssl` aligned (the drift risk ADR-0009 calls out).
- **Test categories proposed:** hermetic config-render test asserting the public location carries `limit_conn`/`limit_rate`; integration test (curl the testfile, assert throttle/connection cap) on a dev host; regression guard that the default `location /` never serves an unlimited large asset; boundary test (many parallel connections capped); failure-injection (missing zone → config test fails closed). Extend the repo's existing nginx-config tests.
- **Minimum-code-for-correctness:** a few lines of nginx template adding the limits to the public download location — no new services, no PHP, no new packages.
- **Success signal:** reliability — the public front door cannot be trivially used as an unlimited bandwidth drain; this protects egress cost and server responsiveness on shared hardware (a measurable abuse-prevention win).

### Architectural Minimization
- **Question the requirement:** "Integrate LibreSpeed" assumes a *web app* is the right delivery. The underlying need is "let someone verify network speed to a server," and a static downloadable file already covers the dominant download-speed question with zero backend.
- **Delete/merge/refactor:** the strongest alternative is to **delete the LibreSpeed backend** (backendless/static removes the PHP surface) or **delete the app entirely** (the existing `/var/www/testfile` already delivers a download test) and instead harden + document the static file. Fold the public-asset limit into the existing default-site template rather than adding a new mechanism.
- **Null option:** doing nothing leaves the public testfile serving **unlimited** — a standing bandwidth-abuse exposure that contradicts the issue's own DOS concern.
- **Minimal design:** add a dedicated public download location (`= /testfile`, plus any future `/speedtest/` static assets) to `template.nginx-site-default` and `-ssl`, applying `limit_conn addr N` + `limit_rate_after`/`limit_rate` via the already-declared `addr` zone. No PHP, no new packages, two template files touched. The LibreSpeed UI is **not** part of the minimal design.

### ADR Adherence
- **ADR-0009 (nginx lightweight reverse proxy):** the rate-limit slice keeps nginx serving static `/var/www` with native limits and adds no app logic or per-app proxy — it **extends** "keep nginx light." A PHP-backed LibreSpeed would **conflict** with ADR-0009 and is therefore not recommended as specified.
- **ADR-0010 (fail2ban — not enabled by default):** the abuse limit uses stateless nginx `limit_conn`/`limit_rate`, not stateful banning — **adheres**.
- **ADR-0015 (defer browser console until packaged + reviewed):** LibreSpeed is the same class of new public web-app surface; this report applies the same gate (defer / static-only until packaged, security-reviewed, and proxied per ADR-0009) rather than a one-pass bolt-on — **extends**.

### Self-Challenge (disproving searches)
- *"The public testfile has no conn/rate limits"* — grepped all nginx templates + the global conf for `limit_rate`/`limit_conn`/`limit_req`: limits appear only in `template.nginx-user`; the default site only declares the zone. **Held.**
- *"Main nginx has no PHP backend"* — grepped nginx templates, `scripts/`, and `etc/` for `fastcgi`/`php-fpm`: none (per-user web is lighttpd+php). **Held.**
- *"The wiki contradicts a branded LibreSpeed"* — re-read the Network Diagnostics page: it cautions that speed tests are unreliable and steers to iperf3, but does not prohibit a provider-hosted endpoint. **Revised to a softer claim** — a reliability *caution/tension* to weigh, not a blocker.

### Recommendation
**SPLIT_FIRST** — split #4 into two work items:

- **Slice A — implement now (small, in-rails):** apply `limit_conn addr N` + `limit_rate_after`/`limit_rate` to the public `/var/www` download location (a dedicated `= /testfile` block) in `template.nginx-site-default` and `-ssl`, reusing the already-declared `addr` zone. This directly satisfies the issue's "restrictions/limits to avoid abuse/DOS" for an asset that is *already* publicly exposed, with no new attack surface. (Extends ADR-0009, adheres ADR-0010.)
- **Slice B — defer / do-not-implement as specified:** do not bolt a PHP-backed LibreSpeed onto the lightweight nginx (conflicts with ADR-0009; matches the ADR-0015 "defer new web-app surface" precedent). If a public "test speed to our datacenter" surface is wanted for marketing/pre-sales, deliver it **backendless/static** under `/var/www/speedtest/` behind the Slice-A limits, after a security/packaging review — that marketing-value-vs-bandwidth-cost call belongs to the maintainers.

**Reasoning:** the issue bundles one clearly-good, cheap safety fix (the requested DOS limit, currently missing on an already-public file) with one contested product feature (a public web app that conflicts with the accepted nginx architecture and the project's published speed-test stance). Splitting lets the safety win land immediately while the feature gets the product decision and review it actually needs.

**Suggested follow-up issue:** *"nginx: apply limit_conn/limit_rate to the public /var/www download asset (testfile)"* — harden the already-public download path with stateless nginx limits, reusing the declared `addr` zone.

**Suggested ADR:** *"Public speed-test surface: static testfile + nginx limits now; defer/gating for a hosted LibreSpeed UI"* — records that the static path is hardened immediately, nginx stays light, and any interactive UI is gated behind packaging + security review (backendless preferred).

— Väinämöinen

**Comment by MagnaCapax** [comment]
## Re-verification (2026-05-31)

Independently re-verified the [2026-05-30 investigation report above](https://github.com/MagnaCapax/PMSS/issues/4) against current source. Every load-bearing finding still holds:

- `scripts/util/update-step2.php` still generates the public 100 MB `/var/www/testfile` on update (regenerated if missing or wrong size).
- `template.nginx-site-default` still declares `limit_conn_zone … zone=addr:10m` at line 1 but applies **no** `limit_conn`/`limit_rate` to `location /` (the `/var/www` public path). Those limits live only in `template.nginx-user`; there is no global `limit_req` in `template.nginx-conf`, and no dedicated `/testfile`/`/speedtest` location. So the public download is still served **uncapped**.
- The main nginx still has **no PHP/FastCGI** wiring.
- ADR-0009, ADR-0010, ADR-0015 are unchanged and remain decisive.

**Nothing has changed since the prior report**, so no new analysis is warranted. This issue is now blocked on a **decision/action, not on further investigation** — the `SPLIT_FIRST` verdict stands:

- **Slice A — actionable now (small, in-rails):** add `limit_conn`/`limit_rate` to a dedicated public download location in `template.nginx-site-default` and `-ssl`, reusing the already-declared `addr` zone. This closes the one concrete gap the issue names ("restrictions and limits … to avoid abuse / DOS"), with no new attack surface. Extends ADR-0009, adheres ADR-0010. Best tracked as its own focused issue.
- **Slice B — product/architecture decision:** an interactive LibreSpeed UI is a new public web-app surface. Keep it off the lightweight nginx (ADR-0009) and gate it the same way the browser-console request (#326) was gated under ADR-0015 — static/backendless only if pursued. Note the public Network Diagnostics page already frames browser speed tests as unreliable.

**Recommendation:** split #4 along this line so the hardening (Slice A) can land immediately and the LibreSpeed-UI question gets the explicit product call it needs — rather than the bundled, underspecified issue being re-picked for repeated investigation.

— Väinämöinen

**Comment by MagnaCapax** [comment]
## Adversarial Review

**Issue:** #4 — Integration of libreSpeed + speed test files, public
**Reviewed:** 2026-06-08
**Verdict:** SHOULD-NOT-IMPLEMENT (as specified) — actionable slice extracted to #641

**Why this comment exists (not a re-investigation).** This issue already carries two thorough investigation reports (2026-05-30, 2026-05-31) that converged on "split it." This is a different lens — an adversarial pass — and it exists for two concrete reasons: (1) the prior public recommendation contains a **latent design error** that would produce broken or no-op behaviour if implemented literally, and (2) the recommended split has been recommended repeatedly but never executed, so I have executed it (#641). I am not re-deriving the prior facts; I independently re-verified them and they hold.

### Adversarial Findings

- **[CORRECTION] `limit_rate` is the wrong tool for a speed-test file, and copied verbatim it is a no-op.** The prior recommendation was to apply `limit_conn` **+ `limit_rate`/`limit_rate_after`** "reusing the proven `template.nginx-user` pattern." Two problems. (a) A speed-test file exists to measure maximum throughput; `limit_rate` throttles per-connection bandwidth and would report misleadingly low speeds — actively defeating the feature and risking a "your network is slow" misread by prospects. (b) The per-user pattern uses `limit_rate_after 100m` (and `1024m`); the testfile is **exactly** 100 MiB (`104857600` bytes, `update-step2.php`), so with `limit_rate_after 100m` the entire file transfers **before** the limit engages — the directive does nothing. The correct tool is `limit_conn` alone (caps concurrent connections per IP, preserves measured speed). This is the core adversarial value of this pass; #641 specifies the corrected design.

- **[CORRECTION] The "edit all three default-site templates" guidance is imprecise.** `setup.php:85,131` reads `template.nginx-site-default` (which contains **both** the `:80` and `:443 ssl default_server` server blocks) and substitutes the SSL snippet into `||SSL_SETTINGS_CONFIGURED_HERE||`. The `-ssl` and `-ssl-lets-encrypt` files are cert/protocol snippets with **no** server or location context. The `location` block therefore goes into **one file, two server blocks** — not "all three templates."

- **[OPERATOR_HANDOFF_ANTIPATTERN — cleared] The split was recommended four times and never executed.** Memory shows this issue has been re-selected repeatedly; every pass recommended creating the child issue and none did, because the recommending role could not. Recommending the same action a fifth time would be the loop. Executing it (#641) is the closure.

**Big Questions:**
1. **Worst case if implemented as specified:** a PHP backend (garbage/empty/getIP) is wired onto the main nginx front door to satisfy LibreSpeed → new executable surface on every host's public `:80`/`:443` → a public, unauthenticated upload/compute endpoint becomes a fleet-wide attack and abuse surface → exploited or abused before anyone notices → emergency fleet-wide config rollback. Distance from "correct" is large: the architecture (ADR-0009) explicitly forbids app logic on nginx.
2. **To guarantee maximum damage:** apply an aggressive `limit_rate` to the testfile so every speed test reports terrible numbers, *and* bolt unauthenticated PHP onto the public front door. The proposed-as-written path drifts toward both.
3. **Regret in 6 months:** a PHP speed-test backend on the public front door becomes a standing per-host maintenance + CVE-tracking burden that the lightweight-nginx design was specifically built to avoid.
4. **Removal inversion:** the existing `/var/www/testfile` is undocumented and unlinked; if someone proposed removing it we would not fight hard — which tells you the "speed test file" need is weakly held, and weakens the case for investing in a full UI.
5. **Dumbest requirement:** "Integrate LibreSpeed" assumes a web *app* is the delivery mechanism. The underlying need is "let someone verify network speed to our server," and a static file already answers the dominant download-speed question with zero backend.
6. **Deletion opportunity:** instead of building the app, keep the static file, add the one missing `limit_conn` (#641), and delete the LibreSpeed-app idea from scope unless a product decision revives it static/backendless.

### Caller Analysis

This change is additive (no functionality removed), but the *placement* has a blast radius worth stating for the implementer:
- Attaching any limit to `location /` (rather than an exact-match `location = /testfile`) would apply it to **every** static asset under `/var/www`: the error pages (`error-40x/502.html`), `index`, and `/401_images/`. Consumers of those paths are every error response and the default vhost. The exact-match `location = /testfile` confines the limit to the testfile only — this is why #641 specifies exact match.
- The `addr` zone (`template.nginx-site-default:1`) is in the http context (the file is included via `sites-enabled`), so `limit_conn addr N` inside a location is valid with no new zone — the same construct `template.nginx-user` already uses.

### Investigation Results

- **Affected files:** `etc/seedbox/config/template.nginx-site-default` (both server blocks); generation via `scripts/lib/nginxConfig/setup.php` → `sites-available/default`. Testfile lifecycle in `scripts/util/update-step2.php`.
- **Test coverage:** `scripts/lib/tests/development/ErrorPageTemplateTest.php` asserts per-server-block symmetry (locations appear twice). A `location = /testfile` test should mirror that (count == 2, carries `limit_conn`).
- **Git context:** the testfile and the `addr` zone are operator-maintained (carried through refactors since the 2023 import) — Chesterton fence: intentional, not dead code. No `limit_req` exists anywhere in the tree.
- **Constraints:** ADR-0009 (nginx must not implement app logic / per-app routing — static + native limits adhere; a PHP backend conflicts); ADR-0015 (defer new public web-app surfaces until packaged + security-reviewed — direct precedent for the LibreSpeed UI); ADR-0010 (no default fail2ban — use stateless nginx limits).

### Recommendations

- **Slice A (hardening):** implement **#641** — `limit_conn addr N` on an exact-match `location = /testfile`, both server blocks, reusing the declared zone. No `limit_rate`.
- **Slice B (LibreSpeed UI):** do not bolt a PHP-backed app onto the main nginx. Whether to offer a public speed-test UI **at all** is a product/architecture decision reserved to the maintainers; if pursued, it must be static/backendless and clear the ADR-0015 gate (packaging + security review). This verdict does not foreclose that product decision — it only rejects the bolt-on-PHP approach.
- **For #4 itself:** close in favour of #641 (Slice A) + a separate product decision for the UI, or re-scope #4 to the UI question only. Labeling `should-not-implement` so the build pipeline stops re-selecting the bundled issue for implementation.

### Worth-Implementing Analysis

- **Cost:** Slice A — low (a few template lines + one test; zone already declared). Slice B as specified — high (PHP surface on the public front door, per-host maintenance, security review).
- **Benefit:** Slice A — closes a real, operator-requested abuse gap on an already-public asset without breaking the test. Slice B as specified — a friendlier UI than a raw file, but at architectural cost the project has explicitly chosen not to pay.
- **Verdict rationale:** the salvageable, in-rails, operator-requested element is Slice A (extracted to #641, PROCEED-class). The issue *as specified* (LibreSpeed app on main nginx) conflicts with ADR-0009/0015 → SHOULD-NOT-IMPLEMENT as specified.
- **Alternative:** static/backendless speed-test surface behind the #641 limits, as a future product decision — not build-agent work.

### Anti-Pattern Hunt

- **REGEX_FOR_SEMANTICS:** PASS — no semantic/intent judged by pattern matching.
- **PREMATURE_OPTIMIZATION:** PASS — Slice A is operator-requested security hardening of a shipped, exposed asset, not a perf optimization of an unproven feature; base feature (testfile) exists and has shipped for years. (Recommend adding the missing test.)
- **OPERATOR_HANDOFF_ANTIPATTERN:** PASS — the closable work (Slice A) is *not* handed off; it is extracted and specified (#641). Only the genuine product/architecture decision (build a public UI at all?) is left to maintainers, which is legitimately theirs.
- **LAZY_HALT_AFTER_AUTHORIZATION:** PASS — investigation proceeded to caller/assembly/test detail and an executed action.
- **FALSE_COMPLETION_CLAIM:** PASS — nothing claimed done that is not; #641 is filed, not implemented.
- **FABRICATED_AUTHORITY_OR_PATH:** PASS — every cited file, line, and ADR verified against source this session.
- **FAIL_OPEN_SAFETY_BEHAVIOR:** PASS — if the `addr` zone were missing, nginx refuses to load the config (fail-closed), not silently open.
- **DRY_RUN_THEATER:** PASS — N/A.
- **INVESTIGATION_LOOP_THEATER:** PASS (with justification) — the risk is real (5th touch on a converged issue); cleared because this pass adds net-new signal (the `limit_rate` correction) and takes the loop-breaking action (#641). Absent both, the correct call would have been a silent skip.
- **LACK_OF_INVESTIGATION:** PASS — counter-signals to "uncapped testfile is a problem": (a) git history of the limit — queried, no evidence a limit was ever present and removed; the asset is operator-maintained; (b) real abuse incidents — not queryable from source in this read-only pass (no production log access), stated honestly; the gap is a latent exposure, not a confirmed active fire.

### Merit Check — Tier A

1. **Concrete present demand.** Operator, in the issue body: *"Some restrictions and limits needs to be put in place however to avoid abuse / DOS type behavior."* That is concrete demand for the Slice A limit. The LibreSpeed UI is a 2024 enhancement aspiration with no recurring pain attached — theoretical demand.
2. **Function.** Slice A: adds `limit_conn addr N` to an exact-match `location = /testfile` in both default-site server blocks, capping concurrent downloads per client IP. Slice B: would add an interactive browser speed-test app (download/upload/ping) to the public front door.
3. **Success metric.** Slice A: `nginx -t` passes and a per-IP parallel-connection flood against `/testfile` is capped at N while a single-stream download still measures full link speed — both observable on a dev host independent of the code's own logs. Slice B: no independent success metric beyond "a UI exists."
4. **Catastrophic if implemented.** Slice B as specified: PHP wired onto public nginx → unauthenticated compute/upload endpoint fleet-wide → abused/exploited → detected late (no per-endpoint monitoring) → emergency fleet rollback. Slice A done wrong (`limit_rate`): speed test reports throttled numbers → prospects conclude the network is slow → lost conversions, surfacing only as vague sales attrition. #641 removes the `limit_rate` foot-gun.
5. **Catastrophic if NOT implemented.** Slice A skipped: the public 100 MiB file stays downloadable with unlimited per-IP concurrency — a latent egress/connection-amplification surface on egress-metered hardware, and the operator's stated requirement stays unmet. Slice B skipped: no observable harm — the static file already answers the dominant download-speed question. The asymmetry (Slice A worth doing, Slice B not-as-specified) is the verdict.
6. **Deletion alternative.** Keep the static file, add the one `limit_conn`, and delete the LibreSpeed-app scope. The best part is no part: a backend the architecture forbids is the part to remove.
7. **Cost.** (a) Build: Slice A ~a few lines + one test; Slice B high (PHP surface). (b) Maintenance/yr: Slice A — revisit N if legitimate multi-stream clients are throttled; Slice B — recurring CVE tracking + per-host upkeep of a PHP app. (c) Blast radius if broken: Slice A — confined to `/testfile` via exact match; mis-scoped to `location /` it would touch all `/var/www` static assets fleet-wide. (d) Undo: Slice A — clean template revert (two-way door); Slice B — removing a shipped public endpoint is messier.
8. **Dependency blast (Hyrum).** The default vhost serves error pages, `index`, and `/401_images/` from `/var/www`; an over-broad limit would change their behaviour. The `addr` zone is shared with the per-user proxy blocks (`template.nginx-user`) — adding a *consumer* of the zone is safe; do not alter the zone definition.
9. **Requirements challenge (Musk Step 1).** The dumbest implied requirement is that abuse protection means `limit_rate` (bandwidth throttle). For a speed-test file that is self-defeating; the requirement should be "cap concurrency/request-rate, never per-connection speed." #641 encodes the corrected requirement.

**Tier B — Goodhart (rate-limit/policy trigger):** optimizing for "we have a DOS limit" can create false confidence — `limit_conn` is per-IP and does not stop distributed (many-IP) abuse; and an over-tight `limit_rate` would *game the speed-test metric itself* (artificially low readings that look like a slow network). #641 states this explicitly so the limit is not mis-sold as complete DOS protection.

### Decomposition Targets

- **[REFACTOR] `etc/seedbox/config/template.nginx-site-default`** — the `:80` and `:443` server blocks duplicate their entire location set (error pages, `/401_images/`, `location /`). New shared locations (like `= /testfile`) must be added twice and kept symmetric by hand; a single shared include for common locations would remove this drift class (the same drift ADR-0009 calls out for proxy params).
- **[REFACTOR] `/var/www/testfile` discoverability** — the file is generated every update but undocumented and unlinked. Either document/link it (so the existing capability is usable) or reconsider regenerating a 100 MiB random file on every update. Separate from this issue; noted for the backlog.

— Väinämöinen
